torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -47,9 +47,21 @@ void debugPrint(const TensorTypePtr& type) {
     for (int i = 0; i < rank; i++) {
       const auto& shape_symbol = sizes.value()[i];
       if (shape_symbol.is_static()) {
+#if defined(__APPLE__)
+        // int64_t format: mac int64_t=long long, %lld
+        printf("%lld, ", shape_symbol.static_size());
+#else
+        // int64_t format: linux int64_t=long int, %ld
         printf("%ld, ", shape_symbol.static_size());
```

Comment:
Using `PRId64` here would probably be less wordy

Reply:
`PRId64` is really ugly, wouldn't it be better to just cast it to one format

Reply:
@malfet PRId64 macro can be also a option, for casting, do you mean that we forced to change into %ld compatible type ? not sure if it's more suitable

Reply:
@llv22 `PRId64` is a macro defined in `cinttypes`, and sometimes is not properly handled by older compilers.
Entire `debugPrint()` function doesn't look right, let me try to modernize it.

Reply:
How about something like the following: https://github.com/pytorch/pytorch/pull/46994

Reply:
@malfet in this case, perhaps we convert uptward to "long long int", then we can use %lld. ^_^, anyway, just for printing purpose, don't need to introduce too much complexity.

Reply:
@malfet haven't seen your comment. std::stringstream sizes_s can successfully avoid such issue for finding correct format, ^-^